typed/parser.go

```diff
@@ -102,12 +102,28 @@ func (p ParseableType) FromYAML(object YAMLObject) (*TypedValue, error) {
 	return AsTyped(value.NewValueInterface(v), p.Schema, p.TypeRef)
 }
 
-// FromUnstructured converts a go interface to a TypedValue. It will return an
+// FromUnstructured converts a go "interface{}" type, typically an
+// unstructured object in Kubernetes world, to a TypedValue. It returns an
 // error if the resulting object fails schema validation.
+// The provided interface{} must be one of: map[string]interface{},
+// map[interface{}]interface{}, []interface{}, int types, float types,
+// string or boolean. Nested interface{} must also be one of these types.
 func (p ParseableType) FromUnstructured(in interface{}) (*TypedValue, error) {
 	return AsTyped(value.NewValueInterface(in), p.Schema, p.TypeRef)
 }
 
+// FromStructured converts a go "interface{}" type, typically an structured object in
```

Comment:
Should you make it clear in the doc that this should be a pointer?

Reply:
Yes, I missed this one. Fixed by applying same text I had added to NewValueInterface about pointers.